index.rst

```diff
@@ -42,20 +42,151 @@
+The group ID will be provided to each script by the ScriptQueue and will be unique for that execution of the script.
+
+Since DM software always mediates data access via a Data Butler this leads to a requirement that raw data must always be retrievable from a butler with a Data ID solely containing the obsid, and optionally, the detector name or group name.
+:cite:`DMTN-111` also states that some event must be issued whenever new data have been ingested and that that event must contain the obsid.
+
+.. [*] In :cite:`LTS-836` the group ID is called a "transaction ID" since it is associated with a single queue transaction.
+
+Processing Data as it Arrives
+=============================
+
+All triggered processing should be done through the "Gen3" middleware pipeline system consisting of a pipeline of``PipelineTask``.
+Each trigger will need to specify at minimum the name of the Pipeline to execute and the data ID that should be processed.
+Additionally it may be necessary to provide an explicit config file by name.
+Using standard change-controlled pipeline configurations is preferable to scripts trying to send override configuration on-the-fly since it provides a self-contained way to reproduce the processing.
+
+For a Pipeline to be executed the data ID needs to be converted to a quantum graph suitable for passing to the activator.
+The data ID would usually be the obsid for the data just acquired.
+Furthermore the group ID could be used to obtain all data that have previously been taken related to this observation.
 
 .. note::
 
-   **This technote is not yet published.**
+  No LSST rapid analysis pipelines are required to co-add data incrementally.
+  Pipelines should not co-add data as it arrives and should always defer co-addition until the final observation in the group has been taken.
+
+Single frame processing is the simplest mode whereby the products from the processing are not to be combined with subsequent observations.
+The more complex scenario relates to processing that requires co-addition where the co-addition can only occur at the end of data acquisition.
+
+One proposal for this is to have a chained pipeline approach.
+As data arrives they are processed with the nominated single frame processing variant.
+On completion they register themselves as inputs to a combination pipeline.
+This pipeline does not run automatically but waits to be told that all data have arrived.
+If an observation fails to reduce properly, or fails to pass some metrics threshold it will not register with the combination pipeline.
+
+When all expected data have arrived and been processed on their own, the final combination pipeline will be unblocked and will run to completion.
+Events will be issued on the completion of a single frame processing and on the completion of the combination pipeline.
+Metrics will also be published where appropriate during processing although the mechanism for this has not been defined.
```

Comment:
The nominal mechanism for this is the "Telemetry Gateway", a SAL-speaking service that can observe messages (notionally RabbitMQ, possibly Kafka) and Butler dataset outputs and the state of, e.g., the Archiver Redis and translate into the LSE-72-required events and telemetry.  Alternate designs could be considered.

Reply:
I'm fine with *a* telemetry gateway so long as it's running at the summit.

Reply:
The Telemetry Gateway is expected to run at the Base.  This could be re-thought, but note that many of the things that it is likely to observe or listen to will come from the Base or even NCSA.

Reply:
Couldn't we have two of them?